Add canEditData prop to determine whether shipping data could be edited or not

jeffersontuc opened the pull request:
#### What problem is this solving?

As the title says.

<!--- What is the motivation and context for this change? -->

#### How should this be manually tested?

- Proceed to [Workspace](https://fixskeleton--checkoutio.myvtex.com/cart).
- Verify that edit postal code button isn't available.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] [Linked this PR to a Clubhouse story (if applicable).](https://app.clubhouse.io/vtex/story/27047/remover-a-op%C3%A7%C3%A3o-de-alterar-cep-no-carrinho-quando-usu%C3%A1rio-estiver-identificado)
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

<!--- Add a ✔️ where applicable -->
✔️ | Type of Change
---|---
_ | Bug fix <!-- a non-breaking change which fixes an issue -->
✔️| New feature <!-- a non-breaking change which adds functionality -->
_ | Breaking change <!-- fix or feature that would cause existing functionality to change -->
_ | Technical improvements <!-- chores, refactors and overall reduction of technical debt -->

jeffersontuc committed:
Add canEditData prop

marcoskwkm reviewed (changes requested):
I added [some items](https://fixskeleton--checkoutio.myvtex.com/cart/add?sku=31&sku=36&sku=33&sku=250&sku=17) to cart, then clicked "View delivery options" and the component crashed:
```
Uncaught TypeError: Cannot read property 'postalCode' of null
    at EstimateShipping (EstimateShipping.tsx:45)
```
Doesn't seem to reproduce in `master`. Seems to be related to this PR, so I'm requesting changes so you can investigate.

jeffersontuc commented:
> I added [some items](https://fixskeleton--checkoutio.myvtex.com/cart/add?sku=31&sku=36&sku=33&sku=250&sku=17) to cart, then clicked "View delivery options" and the component crashed:
> 
> ```
> Uncaught TypeError: Cannot read property 'postalCode' of null
>     at EstimateShipping (EstimateShipping.tsx:45)
> ```
> 
> Doesn't seem to reproduce in `master`. Seems to be related to this PR, so I'm requesting changes so you can investigate.

The `selectedAddress` field comes as `null` from our graphql and, since I removed the `deliveryOptions.length > 0` verification, it started giving this error. To fix this I'm gonna add a `selectedAddress` verification.

jeffersontuc committed:
Add selectedAddress verification
